delta/sourcing-psql/src/main/scala/ch/epfl/bluebrain/nexus/delta/sourcing/query/SelectFilter.scala

```diff
@@ -0,0 +1,34 @@
+package ch.epfl.bluebrain.nexus.delta.sourcing.query
+
+import ch.epfl.bluebrain.nexus.delta.rdf.IriOrBNode.Iri
+import ch.epfl.bluebrain.nexus.delta.sourcing.model.Tag
+import doobie.util.fragment.Fragment
+
+/**
+  * Contains the information that can be used for filtering when streaming states
+  * @param types
+  *   types that a view is indexing
+  * @param tag
+  *   tag that a view is indexing
+  */
+case class SelectFilter(types: Set[Iri], tag: Tag) {
+
+  /** The types as a postgres array */
+  def typeSqlArray: Fragment =
+    Fragment.const(s"ARRAY[${types.map(t => s"'$t'").mkString(",")}]")
+
+}
+
+object SelectFilter {
+
+  /** All types with specified tag */
+  val tag: Tag => SelectFilter = SelectFilter(Set.empty, _)
+
+  /** All types with latest tag */
+  val latest: SelectFilter = SelectFilter(Set.empty, Tag.Latest)
+
+  /** All types with specified tag if it exists, otherwise latest */
+  val tagOrLatest: Option[Tag] => SelectFilter =
```

Comment:
Personally I find this less clear than if you just use the `tag` method and deal with the option in the caller

Reply:
Wouldn't that make the tag optional in the case class? I don't think it should be optional there